feat: add support for system picker in setDisplayMediaRequestHandler (#43264)
electron/electron: docs/api/desktop-capturer.md

  const mainWindow = new BrowserWindow()

  session.defaultSession.setDisplayMediaRequestHandler((request, callback) => {

>     // If we should use the system picker
>     // Note: this is currently experimental

Reviewer: ```suggestion
    // If we should use the MacOS system picker
    // Note: this is currently experimental
```

Reviewer: Talked with Sam about this; this could technically also be used on Linux and Windows in the future, so we don't need to cal it out as MacOS specific 😄

[`navigator.mediaDevices.getUserMedia`]: https://developer.mozilla.org/en/docs/Web/API/MediaDevices/getUserMedia
[`systemPreferences.getMediaAccessStatus`]: system-preferences.md#systempreferencesgetmediaaccessstatusmediatype-windows-macos

> ### `desktopCapturer.isDisplayMediaSystemPickerAvailable()` _Experimental_

Reviewer: ```suggestion
### `desktopCapturer.isDisplayMediaSystemPickerAvailable()` _Experimental_  _macOS_
```

Returns `Boolean`, whether or not requesting desktop content via

> the system picker is supporter on this platform.

Reviewer 2: ```suggestion
the system picker is supported on this platform.
```

Currently this will only return `true` on macOS 14.4 and higher. When
true you should respect this value and use `systemPickerVideoSource` as

> otherwise the OS may present scary warning dialogs to your users.

Reviewer 2: Does this mean that we'll recommend to avoid using `desktopCapturer.getSources` in general on macOS starting with macOS 15?

Author: That is correct, but don't feel comfortable documenting that till we've tested this new API a bit (you may notice I flagged it as experimental)

Reviewer 2: Sounds good, will keep that in mind!

## Properties

> ### `desktopCapturer.systemPickerVideoSource` _Experimental_ _Readonly_

Reviewer: ```suggestion
### `desktopCapturer.systemPickerVideoSource` _Experimental_ _Readonly_ _macOS_
```

A `DesktopCapturerSource` property that should be used in conjunction with

> `sesssion.setDisplayMediaRequestHandler` to use the system picker instead

Reviewer 2: ```suggestion
[`session.setDisplayMediaRequestHandler`](./session.md#sessetdisplaymediarequesthandlerhandler) to use the system picker instead
```

typo + add link
